### PR TITLE
[tmva][sofie] Add RSofieReader class in TMVA

### PR DESCRIPTION
**Pull request by lmoneta**
Add a new class in TMVA , RSofieReader which giving an input model in ONNX, Keras or PyTorch format, compiles with the interpreter and provides a member function `Compute` to evaluate the model.

### CODE REVIEW

File: tmva/tmva/inc/TMVA/RSofieReader.hxx [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+         }
+         parserCode += "TMVA::Experimental::SOFIE::RModel model = SOFIE::PyKeras::Parse(\"" + path + "\"); \n";
+      }
+      else if (type == kPt) {
+         // use PyTorch direct parser
+         if (gSystem->Load("libPyMVA") < 0) {
+            Error("RSofieReader","Cannot use SOFIE with PyTorch since libPyMVA is missing");
+            return;
+         }
+         parserCode += "TMVA::Experimental::SOFIE::RModel model = SOFIE::PyTorch::Parse(\"" + path + "\"); \n";
```
Comment (sanjibansg) [inline tmva/tmva/inc/TMVA/RSofieReader.hxx:122]: This might not work as expected. The PyTorch parser for SOFIE requires the shapes of the input tensors.

**Review by sanjibansg — COMMENTED** [review]
Nice implementation! This will unify the parsing process and will make it much simpler.

File: tmva/tmva/inc/TMVA/RSofieReader.hxx [hunk 2]
```diff
@@ -0,0 +1,243 @@
+/**********************************************************************************
+ * Project: ROOT - a Root-integrated toolkit for multivariate data analysis       *
```
[the remark this answers is not in the document]
Comment (lmoneta) [reply tmva/tmva/inc/TMVA/RSofieReader.hxx:2]: This is a standard header file of TMVA, I think is fine

**Review by sanjibansg — APPROVED** [review]
LGTM!
Thanks for the implementation, this will be very helpful for SOFIE.

**Pull request merged**